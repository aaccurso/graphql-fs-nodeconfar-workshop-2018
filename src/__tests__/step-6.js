const { graphql } = require('graphql');
const schema = require('../schema');
const { listDir, listDirRecursive } = require('../queries');

describe('step 6', () => {
  it('should query list dir', async () => {
    expect(await graphql({ schema, source: listDir })).toMatchSnapshot();
  });

  it('should query list dir recursively', async () => {
    expect(await graphql({ schema, source: listDirRecursive })).toMatchSnapshot();
  });
});
