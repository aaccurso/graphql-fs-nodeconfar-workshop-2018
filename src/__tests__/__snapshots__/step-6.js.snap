// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`step 6 should query list dir 1`] = `
Object {
  "data": Object {
    "ls": Array [
      Object {
        "name": "Father",
        "type": "Dir",
      },
      Object {
        "name": "Marge_Simpson.png",
        "type": "File",
      },
      Object {
        "name": "Mother",
        "type": "Dir",
      },
      Object {
        "name": "Patty_Bouvier.png",
        "type": "File",
      },
      Object {
        "name": "Selma_Bouvier.png",
        "type": "File",
      },
    ],
  },
}
`;

exports[`step 6 should query list dir recursively 1`] = `
Object {
  "data": Object {
    "ls": Array [
      Object {
        "ls": Array [
          Object {
            "name": "Father",
            "type": "Dir",
          },
          Object {
            "name": "Homer_Simpson.png",
            "type": "File",
          },
          Object {
            "name": "Mother",
            "type": "Dir",
          },
        ],
        "name": "Father",
        "type": "Dir",
      },
      Object {
        "name": "Marge_Simpson.png",
        "type": "File",
      },
      Object {
        "ls": Array [
          Object {
            "name": "Father",
            "type": "Dir",
          },
          Object {
            "name": "Marge_Simpson.png",
            "type": "File",
          },
          Object {
            "name": "Mother",
            "type": "Dir",
          },
          Object {
            "name": "Patty_Bouvier.png",
            "type": "File",
          },
          Object {
            "name": "Selma_Bouvier.png",
            "type": "File",
          },
        ],
        "name": "Mother",
        "type": "Dir",
      },
      Object {
        "name": "Patty_Bouvier.png",
        "type": "File",
      },
      Object {
        "name": "Selma_Bouvier.png",
        "type": "File",
      },
    ],
  },
}
`;
